src/components/misc/zoomer.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+    <div @wheel="onWheel($event)" v-hammer:pinch="onPinch">
+        <slot></slot>
+    </div>
+</template>
+
+<script lang="ts">
+/*
+ * This component handles zooming requests (ctrl-mousewheel and hopefully pinching in the future) properly,
```

Comment:
Yes, this isn't smooth at all. I'm not sure how this can be improved. I think scroll bars make a lot of sense when there's scrolling or panning. If this were a Windows app, I would add two scroll bars and handle all the events myself (scrolling to translate, panning to updating the scrollbar thumb location). I don't know if we can do that in a browser.